modules/docs/markdown/01-overview/05-sharing-specs.md

```diff
@@ -151,17 +151,61 @@ libraryDependencies += "organisation" % "artifact" % "version" % "smithy4s,compi
 ### Mill
 
 ```scala
-
 def compileAndCodegenDeps = T(Agg(ivy"organisation:artifact:version"))
 def ivyDeps = T(super.ivyDeps() ++ compileAndCodegenDeps())
 def smithy4sIvyDeps = T(super.smithy4sIvyDeps() ++ compileAndCodegenDeps())
-
 ```
 
 ### Consequence
 
 Because the upstream usage of Smithy4s will have resulted in the creation of metadata tracking the namespaces that were already generated, the "local" Smithy4s code-generation will automatically skip the generation of code that should not be generated again.
 
+## Artifacts containing Smithy4s generated code : dependency tracking
+
+When packaging a project/module via SBT or Mill, Smithy4s adds a line to the Jar manifest of the project, informing downstream projects of library dependencies that may have been used during the code-generation of this project/module.
+
+This information is used automatically by downstream usage of Smithy4s, which automatically pulls additional jars that would be specified
+in this bit of metadata.
+
+So, for instance, if you have
+
+```scala
+lazy val upstream = (project in file("foo"))
+  .enablePlugins(Smithy4sCodegenPlugin)
+  .settings(
+    organization := "foobar",
+    version := "0.0.1",
+    libraryDependencies ++= Seq(
+      "software.amazon.smithy" % "smithy-aws-iam-traits" % "1.14.1" % Smithy4s
+    )
+  )
+```
+
+and publish this project to an artifact repository, the Jar manifest will contain the following line :
+
+```
+smithy4sDependencies: software.amazon.smithy:smithy-aws-iam-traits:1.14.1
+```
+
+Using this artifact in a downstream project, for instance with :
+
+```scala
+lazy val downstream = (project in file("foo"))
+  .enablePlugins(Smithy4sCodegenPlugin)
+  .settings(
+    libraryDependencies ++= Seq(
+      // compile/runtime dependency that contains Smithy4s-generated code but doesn't contain smithy files
+      "foobar" % "upstream" % "0.0.1"
```

Comment:
```suggestion
      "foobar" % "upstream" % "0.0.1" % Smithy4s
```

Not sure, but seems like the smithy4s would be needed here?

Reply:
Nope, that's the whole point of this PR. The library itself doesn't contain smithy code, but was generated using smithy4s, and therefore has a reference to the artifacts that it pulled for its own code-generation.

Reply:
if it contains generated code it should be considered a Scala dependency, right? so probably `%%` to account for the suffix. Seeing this I'm wondering if the whole thing will work on non-JVM platforms...

Reply:
non-JVM Scala libraries are still packaged in jars, and still carry manifests

Reply:
I do wonder if we shouldn't just use the whole (upstream) classpath for code-generation. I think this may be more intuitive and closer to users' mental models. WDYT ?